drivers/led/issi/is31fl3729-simple.c

```diff
@@ -0,0 +1,223 @@
+#    endif
+#endif
+
+    for (int i = 0; i < IS31FL3729_LED_COUNT; i++) {
+        is31fl3729_set_scaling_register(i, 0xFF);
+    }
+
+    is31fl3729_update_scaling_registers(IS31FL3729_I2C_ADDRESS_1, 0);
+#if defined(IS31FL3729_I2C_ADDRESS_2)
+    is31fl3729_update_scaling_registers(IS31FL3729_I2C_ADDRESS_2, 1);
+#    if defined(IS31FL3729_I2C_ADDRESS_3)
+    is31fl3729_update_scaling_registers(IS31FL3729_I2C_ADDRESS_3, 2);
+#        if defined(IS31FL3729_I2C_ADDRESS_4)
+    is31fl3729_update_scaling_registers(IS31FL3729_I2C_ADDRESS_4, 3);
+#        endif
+#    endif
+#endif
+}
+
+void is31fl3729_init(uint8_t addr) {
+    // In order to avoid the LEDs being driven with garbage data
+    // in the LED driver's PWM registers, shutdown is enabled last.
+    // Set up the mode and other settings, clear the PWM registers,
+    // then disable software shutdown.
+
+    // Set Pull up & Down for SWx CSy
+    is31fl3729_write_register(addr, IS31FL3729_REG_PULLDOWNUP, IS31FL3729_PULLDOWNUP);
+
+    // Set Spread Spectrum Register if applicable
+    is31fl3729_write_register(addr, IS31FL3729_REG_SPREAD_SPECTRUM, IS31FL3729_SPREAD_SPECTRUM);
+
+    // Set PWM Frequency Register if applicable
+    is31fl3729_write_register(addr, IS31FL3729_REG_PWM_FREQUENCY, IS31FL3729_PWM_FREQUENCY);
+
+    // Set Golbal Current Control Register
+    is31fl3729_write_register(addr, IS31FL3729_REG_GLOBALCURRENT, IS31FL3729_GLOBALCURRENT);
+
+    // Set to Normal operation
+    is31fl3729_write_register(addr, IS31FL3729_REG_CONFIGURATION, IS31FL3729_CONFIGURATION);
+
+    // Wait 10ms to ensure the device has woken up.
+    wait_ms(10);
+}
+
+void is31fl3729_set_value(int index, uint8_t value) {
+    is31fl3729_led_t led;
+    if (index >= 0 && index < IS31FL3729_LED_COUNT) {
+        memcpy_P(&led, (&g_is31fl3729_leds[index]), sizeof(led));
+
+        if (g_pwm_buffer[led.driver][led.v] == value) {
+            return;
+        }
+
+        g_pwm_buffer_update_required[led.driver] = true;
+        g_pwm_buffer[led.driver][led.v]          = value;
+    }
+}
+
+void is31fl3729_set_value_all(uint8_t value) {
+    for (int i = 0; i < IS31FL3729_LED_COUNT; i++) {
+        is31fl3729_set_value(i, value);
+    }
+}
+
+void is31fl3729_set_scaling_register(uint8_t index, uint8_t value) {
+    is31fl3729_led_t led;
+    memcpy_P(&led, (&g_is31fl3729_leds[index]), sizeof(led));
+
+    // need to do a bit of checking here since 3729 scaling is per CS pin.
+    // not the usual per single LED key as per other ISSI drivers
+    // only enable them, since they should be default disabled
+    int cs_value = (led.v & 0x0F) - 1;
+
+    g_scaling_registers[led.driver][cs_value] = value;
+
+    g_scaling_registers_update_required[led.driver] = true;
+}
+
+void is31fl3729_update_pwm_buffers(uint8_t addr, uint8_t index) {
+    if (g_pwm_buffer_update_required[index]) {
+        is31fl3729_write_pwm_buffer(addr, g_pwm_buffer[index]);
+    }
+
+    g_pwm_buffer_update_required[index] = false;
```

Comment:
```suggestion
    if (g_pwm_buffer_update_required[index]) {
        is31fl3729_write_pwm_buffer(addr, g_pwm_buffer[index]);

        g_pwm_buffer_update_required[index] = false;
    }
```

Reply:
Thanks, I've just push a new commit to resolved this.